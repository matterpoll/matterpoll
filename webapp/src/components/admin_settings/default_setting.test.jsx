import React from 'react';
import {shallow} from 'enzyme';

import DefaultSetting from '@/components/admin_settings/default_setting';

describe('components/admin_settings/DefaultSettings', () => {
    const baseProps = {
        name: 'test name',
        title: 'test title',
        label: 'test label',
        value: false,
        onChange: jest.fn(),
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<DefaultSetting {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with true option', () => {
        const newProps = {
            ...baseProps,
        };
        newProps.value = true;
        const wrapper = shallow(<DefaultSetting {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
