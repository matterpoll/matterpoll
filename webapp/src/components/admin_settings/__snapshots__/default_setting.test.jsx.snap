// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_settings/DefaultSettings should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12 col-sm-4"
    style={
      Object {
        "marginTop": "6px",
      }
    }
  >
    <strong>
      test title
    </strong>
  </div>
  <div
    className="col-xs-12 col-sm-8"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          defaultChecked={false}
          onClick={[Function]}
          type="checkbox"
        />
        <span>
          test label
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`components/admin_settings/DefaultSettings should match snapshot with true option 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12 col-sm-4"
    style={
      Object {
        "marginTop": "6px",
      }
    }
  >
    <strong>
      test title
    </strong>
  </div>
  <div
    className="col-xs-12 col-sm-8"
  >
    <div
      className="checkbox"
    >
      <label>
        <input
          defaultChecked={true}
          onClick={[Function]}
          type="checkbox"
        />
        <span>
          test label
        </span>
      </label>
    </div>
  </div>
</div>
`;
